src/Module/BaseSearch.php

```diff
@@ -91,14 +91,16 @@ public static function performContactSearch(string $search, string $prefix = '')
 
 		$pager = new Pager(DI::l10n(), DI::args()->getQueryString(), $itemsPerPage);
 
-		if ($localSearch && empty($results)) {
-			$pager->setItemsPerPage(80);
-			$results = Search::getContactsFromLocalDirectory($search, $type, $pager->getStart(), $pager->getItemsPerPage());
-		} elseif (Search::getGlobalDirectory() && empty($results)) {
-			$results = Search::getContactsFromGlobalDirectory($search, $type, $pager->getPage());
-			$pager->setItemsPerPage($results->getItemsPage());
-		} else {
-			$results = new ResultList();
+		if (empty($results)) {
+			if ($localSearch) {
+				$pager->setItemsPerPage(80);
+				$results = Search::getContactsFromLocalDirectory($search, $type, $pager->getStart(), $pager->getItemsPerPage());
+			} elseif (Search::getGlobalDirectory()) {
+				$results = Search::getContactsFromGlobalDirectory($search, $type, $pager->getPage());
+				$pager->setItemsPerPage($results->getItemsPage());
+			} else {
+				$results = new ResultList();
+			}
 		}
```

Comment:
I've been able to fix the same issue with just the following change with no changes to `Core\Search::getContactsFromProbe`
```suggestion
		if (empty($results) || !$results->getTotal()) {
			if ($localSearch) {
				$pager->setItemsPerPage(80);
				$results = Search::getContactsFromLocalDirectory($search, $type, $pager->getStart(), $pager->getItemsPerPage());
			} elseif (Search::getGlobalDirectory()) {
				$results = Search::getContactsFromGlobalDirectory($search, $type, $pager->getPage());
				$pager->setItemsPerPage($results->getItemsPage());
			} else {
				$results = new ResultList();
			}
		}
```

Reply:
The used array fields there should always be filled. If not then there is some deeper issue.

Reply:
Why?

Reply:
Either all fields are there or none. The latter case is handled with the `empty(...)` check. If we experience any issue with some missing fields, then something really is wrong - and we should know about it. Each little warning/notice that we mark with `?? ...` is a potential "let's put a sticker over the warning light" thing.

Reply:
What fields are you talking about?

Reply:
And what about adding an exception? This is a REAL notice instead of unexpected PHP warnings.